tests/dialects/test_tsql.py

```diff
@@ -453,6 +453,8 @@ def test_tsql(self):
             },
         )
 
+        self.validate_identity("SELECT 'BEGIN'")
```

Comment:
This test doesn't seem meaningful– it's not related to the changes in this PR. Consider checking that `SELECT begin` raises for T-SQL, instead.